lib/shoppertrak_api_client.py

```diff
@@ -0,0 +1,161 @@
+                            self._get_xml_str(traffic_xml, "enters")
+                        )
+                        exits = self._cast_str_to_int(
+                            self._get_xml_str(traffic_xml, "exits")
+                        )
+
+                        start_time_str = self._get_xml_str(traffic_xml, "startTime")
+                        start_dt_str = datetime.strptime(
+                            date_str + " " + start_time_str, "%Y%m%d %H%M%S"
+                        ).isoformat()
+
+                        code = self._get_xml_str(traffic_xml, "code")
+                        is_healthy_orbit = code == "01"
+                        if code != "01" and code != "02":
+                            self.logger.warning(
+                                f"Unknown code: '{code}'. Setting is_real to False"
+                            )
+
+                        if is_healthy_orbit or not is_recovery_mode:
+                            rows.append(
+                                {
+                                    "shoppertrak_site_id": site_str,
+                                    "orbit": entrance_val,
+                                    "increment_start": start_dt_str,
+                                    "enters": enters,
+                                    "exits": exits,
+                                    "is_healthy_orbit": is_healthy_orbit,
+                                    "is_recovery_data": is_recovery_mode,
+                                    "poll_date": self.today_str,
+                                }
+                            )
+        return rows
+
+    def _check_response(self, response_text):
+        try:
+            root = ET.fromstring(response_text)
+            error = root.find("error")
+        except Exception as e:
+            self.logger.error(f"Could not parse XML response {response_text}: {e}")
+            raise ShopperTrakApiClientError(
+                f"Could not parse XML response {response_text}: {e}"
+            ) from None
+
+        if error is not None and error.text is not None:
+            # E108 is used when ShopperTrak is busy and they recommend trying again
+            if error.text == "E108":
+                self.logger.info("E108: ShopperTrak is busy")
+                return None
+            else:
+                self.logger.error(f"Error found in XML response: {response_text}")
+                raise ShopperTrakApiClientError(
+                    f"Error found in XML response: {response_text}"
+                )
+        elif len(root.findall(".//traffic")) == 0:
+            self.logger.error(f"No traffic found in XML response: {response_text}")
+            raise ShopperTrakApiClientError(
+                f"No traffic found in XML response: {response_text}"
+            )
+        else:
+            return root
+
+    def _get_xml_str(self, xml, attribute):
+        attribute_str = (xml.get(attribute) or "").strip()
```

Comment:
Is ` or ""` redundant if we just return `None` if the `get` response evaluates to False?

Reply:
Good point -- done!